sos/report/__init__.py

```diff
@@ -153,6 +153,7 @@ def __init__(self, parser, args, cmdline):
         self.opts.merge(self.preset.opts)
         # re-apply any cmdline overrides to the preset
         self.opts = self.apply_options_from_cmdline(self.opts)
+        self._setup_logging()
```

Comment:
I haven't run this yet myself, but my initial thought is that this would create an additional temp file within our temp directory.

Just thinking out loud here - but perhaps what we should instead do here if check the verbosity level after the merging and `apply_options_from_cmdline()` is done, and simply set the level of the logging handlers appropriately (like we do initially in the base component class).

Reply:
Searching the code base (`grep -Inr Handler |grep -v venv`) shows me there is only a [stdout StreamHandler](https://github.com/sosreport/sos/blob/master/sos/component.py#L278), a [log file Stream Handler for sos](https://github.com/sosreport/sos/blob/master/sos/component.py#L271), as well as a [log file StreamHandler for sos_ui](https://github.com/sosreport/sos/blob/master/sos/component.py#L304).

And all the variable files give me the impression that their file names are cached in the Component as variables and when recreating the Handlers one would just need to reuse these file names and ensure that that all previous writes have been flushed to the files and the files are closed. I'll think about how I can put this in code.

Reply:
From my testing, no extra file is created and the patch works well except for:

    020-11-11 14:35:10,109 INFO: [sos.report:setup] effective options now: --batch --build --log-size 100 --plugopts apache.log=on --preset sostestpreset -v


(there should be `-vvv`). But that is a minor issue..

Not sure if the patch can have some side-effect I dont see now, though.